frontend/src/host_orchestrator/orchestrator/userartifacts.go

```diff
@@ -78,6 +83,56 @@ func NewUserArtifactsManagerImpl(opts UserArtifactsManagerOpts) *UserArtifactsMa
 	}
 }
 
+func (m *UserArtifactsManagerImpl) UpdateArtifact(checksum string, chunk UserArtifactChunk) error {
+	if err := createDir(m.RootDir); err != nil {
+		return err
+	}
+	filename := filepath.Join(m.RootDir, checksum)
+	if err := createUAFile(filename, m.Owner); err != nil {
+		return err
+	}
+	if err := writeChunk(filename, chunk); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (m *UserArtifactsManagerImpl) StatArtifact(checksum string) (os.FileInfo, error) {
```

Comment:
I think we need a lock between UpdateArtifact and StatArtifact, if not, a client needs to take care of it. Even though we can assume every client is trusted, we'd better take care of a lock in the server.